app/models/manageiq/providers/embedded_terraform/provider.rb

```diff
@@ -0,0 +1,20 @@
+class ManageIQ::Providers::EmbeddedTerraform::Provider < Provider
```

Comment:
I'm not sure we need an `EmbeddedTerraform::Provider`, this looks like another copy&paste from EmbeddedAnsible

Reply:
Needs review.

Reply:
I removed the included `DefaultTerraformObjects` module.  It looks like the tests are creating provider/managers through `create_in_provider`, and perhaps that's why we have a provider.rb with code that does `ensure_managers`.  I'm not sure how it's expected to work without a provider.

Reply:
I removed `Provider` and fixed the factories and specs.

This was copied&pasted from EmbeddedAnsible, but EmbeddedAnsible seeds a Provider [[ref]](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/seeding.rb#L6) where EmbeddedTerraform does not [[ref]](https://github.com/ManageIQ/manageiq-providers-embedded_terraform/blob/master/lib/manageiq/providers/embedded_terraform.rb#L8-L11).  This meant that these specs did not match the behavior of how the embedded_terraform manager was being created.